File: problems/2273.FindResultantArrayAfterRemovingAnagrams/main_test.go
package problems

import (
	"testing"

	"github.com/stretchr/testify/assert"
)

func TestRemoveAnagrams(t *testing.T) {
	tests := []struct {
		name     string
		words    []string
		expected []string
	}{
		{
			name:     "Example 1",
			words:    []string{"abba", "baba", "bbaa", "cd", "cd"},
			expected: []string{"abba", "cd"},
		},
		{
			name:     "Example 2",
			words:    []string{"a", "b", "c", "d", "e"},
			expected: []string{"a", "b", "c", "d", "e"},
		},
		{
			name:     "Example 3",
			words:    []string{"a", "b", "a"},
			expected: []string{"a", "b", "a"},
		},
		{
			name:     "Empty input",
			words:    []string{},
			expected: []string{},
		},
		{
			name:     "Single word",
			words:    []string{"hello"},
			expected: []string{"hello"},
		},
		{
			name:     "All same anagrams",
			words:    []string{"abc", "bca", "cab", "acb"},
			expected: []string{"abc"},
		},
	}

	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			result := RemoveAnagrams(tt.words)
			assert.Equal(t, tt.expected, result, "The arrays should be equal")
		})
	}
}

File: problems/2273.FindResultantArrayAfterRemovingAnagrams/main.test.ts
import { removeAnagrams } from './main.ts';
import { assertEquals } from '@std/assert/equals';

Deno.test('Example 1: Remove consecutive anagrams', () => {
  const words = ['abba', 'baba', 'bbaa', 'cd', 'cd'];
  const expected = ['abba', 'cd'];
  assertEquals(removeAnagrams(words), expected);
});

Deno.test('Example 2: No anagrams present', () => {
  const words = ['a', 'b', 'c', 'd', 'e'];
  const expected = ['a', 'b', 'c', 'd', 'e'];
  assertEquals(removeAnagrams(words), expected);
});

Deno.test('Example 3: Non-consecutive similar words', () => {
  const words = ['a', 'b', 'a'];
  const expected = ['a', 'b', 'a'];
  assertEquals(removeAnagrams(words), expected);
});

Deno.test('Empty input array', () => {
  const words: string[] = [];
  const expected: string[] = [];
  assertEquals(removeAnagrams(words), expected);
});

Deno.test('Single word in array', () => {
  const words = ['hello'];
  const expected = ['hello'];
  assertEquals(removeAnagrams(words), expected);
});

Deno.test('All consecutive anagrams', () => {
  const words = ['abc', 'bca', 'cab', 'acb'];
  const expected = ['abc'];
  assertEquals(removeAnagrams(words), expected);
});
